rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/AddClassFileMetadata.java

```diff
@@ -48,83 +46,81 @@ public void perform(GraphRewrite event, EvaluationContext context, JavaClassFile
         ExecutionStatistics.get().begin("AddClassFileMetadata.perform()");
         try
         {
-            WindupJavaConfigurationService javaCfgService = new WindupJavaConfigurationService(event.getGraphContext());
+            WindupJavaConfigurationModel javaConfiguration = WindupJavaConfigurationService.getJavaConfigurationModel(event.getGraphContext());
 
+            String absoluteFile = payload.asFile().getAbsolutePath();
+            absoluteFile = FilenameUtils.separatorsToUnix(absoluteFile);
+
+            for (PackageModel excludePackage : javaConfiguration.getExcludeJavaPackages())
+            {
+                String packageAsPath = excludePackage.getPackageName().replace(".", "/");
+                if (absoluteFile.contains(packageAsPath))
+                    return;
+            }
+
+            boolean shouldScan = true;
+            for (PackageModel includePackage : javaConfiguration.getScanJavaPackages())
+            {
+                String packageAsPath = includePackage.getPackageName().replace(".", "/");
+                if (absoluteFile.contains(packageAsPath))
```

Comment:
That won't work if the base for the package is a subdirectory, and I don't want to have special logic for all of the cases where it is not.